### PR TITLE
feat: Rewards Bot Swapping Updates

### PR DESCRIPTION
**Pull request by kruspy**

### CODE REVIEW

File: x/liquidstakeibc/keeper/icq.go [hunk 1]
```diff
@@ -188,3 +192,64 @@ func RewardsAccountBalanceCallback(k Keeper, ctx sdk.Context, data []byte, query
...
+		}
+
+		// build the transfer message to send the rewards to the swapping address
+		msgTransfer := &banktypes.MsgSend{
+			FromAddress: hc.RewardsAccount.Address,
+			ToAddress:   hc.RewardParams.Destination,
+			Amount:      sdk.NewCoins(autocompoundRewards),
+		}
+
+		// execute the ICA transfer transaction
```
Comment (puneet2019) [inline x/liquidstakeibc/keeper/icq.go:226]:
```suggestion
		// execute the ICA msgSend transaction
```

File: x/liquidstakeibc/keeper/icq.go [hunk 2]
```diff
@@ -188,3 +192,64 @@ func RewardsAccountBalanceCallback(k Keeper, ctx sdk.Context, data []byte, query
...
+		_, err = k.GenerateAndExecuteICATx(
```
Comment (puneet2019) [inline x/liquidstakeibc/keeper/icq.go:227]: is the ica Ack/ Timeout handled?

Comment (kruspy) [reply x/liquidstakeibc/keeper/icq.go:227]: if it times out, will retry the next epoch

Comment (puneet2019) [reply x/liquidstakeibc/keeper/icq.go:227]: i see, and how does dydx compounding happen?

Comment (kruspy) [reply x/liquidstakeibc/keeper/icq.go:227]: script will take rewards from the destination address, swap them, and send them to the deposit address directly

Comment (puneet2019) [reply x/liquidstakeibc/keeper/icq.go:227]: deposit address on persistence or host_chain? 
Is there deposit (delegatable or just deposit) entry being added?

Comment (kruspy) [reply x/liquidstakeibc/keeper/icq.go:227]: the flow will be the following:

1. Module queries for the balance of the denom inside RewardParams.
2. Upon response, the max compoundable amount is sent to the destination address inside RewardParams.
3. Job will swap rewards on Dexter (routing them properly).
4. Rewards are sent back to the reward ICA account.
5. Module picks up rewards as normally.

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**